src/peergos/shared/user/fs/MimeTypes.java

```diff
@@ -150,13 +156,30 @@ public static final String calculateMimeType(byte[] start, String filename) {
                 return "text/svg+xml";
             return "text/plain";
         }
-
-        if (filename.endsWith(UserContext.App.Todo.TODO_FILE_EXTENSION))
-            return "application/vnd.peergos-todo";
-
+        Optional<String> cborMimeType = extractCborMimeType(start);
+        if (cborMimeType.isPresent() ) {
+            if (cborMimeType.get().equals(UserContext.App.Todo.TODO_MIME_TYPE) && filename.endsWith(UserContext.App.Todo.TODO_FILE_EXTENSION)) {
+                return UserContext.App.Todo.TODO_MIME_TYPE;
+            }
+        }
         return "application/octet-stream";
     }
 
+    private static Optional<String> extractCborMimeType(byte[] data) {
+        try {
+            CborDecoder decoder = new CborDecoder(new ByteArrayInputStream(data));
```

Comment:
we should be able to do this without any explicit dependency on cbor code, just look at the raw bytes.